### PR TITLE
drivers: nrf_wifi: Restructure to carve out a new nrf70 bus lib

### PR DESCRIPTION
**Pull request by bama-nordic**
The nrf wifi drivers include code related to qspi/spi bus. This work moves bus related files to a new nrf70 bus library which is now independent of the core driver and facilitlates building applications directly using the bus APIs, agnostic to enabling wifi configs.

manifest-pr-skip

### CODE REVIEW

File: modules/nrf_wifi/os/CMakeLists.txt [hunk 1]
```diff
@@ -38,6 +38,7 @@ target_compile_definitions(
   $<$<BOOL:${CONFIG_NRF_WIFI_MGMT_BUFF_OFFLOAD}>:NRF_WIFI_MGMT_BUFF_OFFLOAD>
```
Comment (jukkar) [inline modules/nrf_wifi/os/CMakeLists.txt:38]: What is the point of this last commit, can't you squash it to the first one?

**Comment by jukkar** [comment]
Why are we skipping manifest PR?

File: modules/nrf_wifi/os/CMakeLists.txt (hunk 1 above)
Comment (bama-nordic) [reply modules/nrf_wifi/os/CMakeLists.txt:38]: autosquash did not do this due to saucetag. Fixed now, Thanks!

**Comment by bama-nordic** [comment]
> Why are we skipping manifest PR?

There were non-manifest changes which I wanted to club together - added to https://github.com/nrfconnect/sdk-nrf/pull/19232

**Review by krish2718 — APPROVED** [review]

**Review by rado17 — APPROVED** [review]

**Review by D-Triveni — APPROVED** [review]

**Comment by carlescufi** [comment]
Superseded by #2339

**Pull request closed without merging**